File: app/src/main/java/com/bubelov/coins/dagger/Injector.java
package com.bubelov.coins.dagger;

import android.content.Context;

/**
 * @author Igor Bubelov
 */

public enum Injector {
    INSTANCE;

    private MainComponent mainComponent;

    public void initMainComponent(Context context) {
        mainComponent = DaggerMainComponent.builder().mainModule(new MainModule(context)).build();
    }

    public MainComponent mainComponent() {
        return mainComponent;
    }
}

File: app/src/main/java/com/bubelov/coins/dagger/Injector.kt
package com.bubelov.coins.dagger

import android.content.Context

/**
 * @author Igor Bubelov
 */

object Injector {
    lateinit var appComponent: AppComponent
    private set

    fun init(context: Context) {
        appComponent = DaggerAppComponent.builder().context(context).build()
    }
}